### DISCUSSION TITLE
What can be modified in `~/.kit` folder?

### DISCUSSION THREAD

**Discussion by lannonbr** [johnlindquist/kit#159] [Q&A] [answered]
So I wanted to create a way making "private scripts" where they don't show up in the main tabs, but a `private` tab which you have to go through an `arg()` call before seeing them. With this, I went into modify the various main menu scripts (`kit/main/edit.js`, `kit/main/share.js`, etc) to do a filter on the menu of scripts to exclude ones that start with `private-`.

What I'm curious about is when I grab a new version of Script Kit, will those files be blown away with the new release? I'm curious what is a good pattern if you want to modify the various core scripts that are found in the `.kit` folder?

**Comment by johnlindquist** [comment, marked as answer]
> Related: I'm adding `//Exclude` metadata for scripts you don't want in the main menu.

## Short answer

Nothing. Treat `~/.kit`as an sdk.  Plan on `/.kit` being deleted each update.

## Longer answer:

This week, I'm working on two modes:

1. Default
Kit.app will completely avoid git and bundle the latest version of `~/.kit` which each update. The update will remove the previous `~/.kit` and replace it with the new version


2. Contributor mode
* Kit.app will check if `~/.kit` is a git repo by checking if a `~/.kit/.git` dir exists.
* On update: Kit.app will stash any changes and pull the latest from main

## Why?
There have been too many bugs relying on git:  #129 , #88, #69

**Reply by lannonbr** [reply]
Awesome. At least for my issue, the `// Exclude` metadata should resolve things, but yeah those two modes will be useful if I want to tweak "the SDK"
